src/osd/PGBackend.cc

```diff
@@ -832,11 +832,11 @@ map<pg_shard_t, ScrubMap *>::const_iterator
     // This is automatically corrected in PG::_repair_oinfo_oid()
     assert(oi.soid == obj);
 
-    if (auth_version != eversion_t()) {
-      if (!object_error.has_object_info_inconsistency() && !(bl == auth_bl)) {
-	object_error.set_object_info_inconsistency();
-	error_string += " object_info_inconsistency";
-      }
+    if (first_bl.length() == 0) {
+      first_bl.append(bl);
+    } else if (!object_error.has_object_info_inconsistency() && !(bl == first_bl)) {
```

Comment:
> This test is hopefully selecting the object info with the highest version and the most digests available.

@dzafman i understand this. that's why i raise this concern. because the first one is not necessary the one with the highest version and the with the most digests available.